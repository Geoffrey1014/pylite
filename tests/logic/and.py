if True and True:
    print 'true'
else:
    print 'false'

if True and False:
    print 'true'
else:
    print 'false'

if False and True:
    print 'true'
else:
    print 'false'
    
if False and False:
    print 'true'
else:
    print 'false'

print True and True
print True and False
print False and True
print False and False
