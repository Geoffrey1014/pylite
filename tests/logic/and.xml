lambda
  name __init__
  param_count 0
  local_count 0
  label_count 20
  insn_vector
    load_const c0
    jfalse L1
    load_const c0
    jfalse L1
    load_const c1
    print
    jmp L0
    label L1
    load_const c2
    print
    label L0
    load_const c0
    jfalse L3
    load_const c3
    jfalse L3
    load_const c1
    print
    jmp L2
    label L3
    load_const c2
    print
    label L2
    load_const c3
    jfalse L5
    load_const c0
    jfalse L5
    load_const c1
    print
    jmp L4
    label L5
    load_const c2
    print
    label L4
    load_const c3
    jfalse L7
    load_const c3
    jfalse L7
    load_const c1
    print
    jmp L6
    label L7
    load_const c2
    print
    label L6
    load_const c0
    jfalse L10
    load_const c0
    jtrue L8
    label L10
    load_const c3
    jmp L9
    label L8
    load_const c0
    label L9
    print
    load_const c0
    jfalse L13
    load_const c3
    jtrue L11
    label L13
    load_const c3
    jmp L12
    label L11
    load_const c0
    label L12
    print
    load_const c3
    jfalse L16
    load_const c0
    jtrue L14
    label L16
    load_const c3
    jmp L15
    label L14
    load_const c0
    label L15
    print
    load_const c3
    jfalse L19
    load_const c3
    jtrue L17
    label L19
    load_const c3
    jmp L18
    label L17
    load_const c0
    label L18
    print
    load_const c4
    return
  const_vector
    True
    "true"
    "false"
    False
    None
  guard_vector
  line_number_map 1 1 1 1 2 2 1 1 4 4 1 6 6 6 6 7 7 6 6 9 9 6 11 11 11 11 12 12 11 11 14 14 11 16 16 16 16 17 17 16 16 19 19 16 21 21 21 21 21 21 21 21 21 21 21 22 22 22 22 22 22 22 22 22 22 22 23 23 23 23 23 23 23 23 23 23 23 24 24 24 24 24 24 24 24 24 24 24 1 1
