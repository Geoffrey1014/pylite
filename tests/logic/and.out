true
false
false
false
True
False
False
False
